cmd/engine/engine.go

```diff
@@ -178,6 +178,30 @@ func doCmd(_ *cobra.Command, _ []string) (err error) {
 	assessmentService = service_assessment.NewService()
 	evidenceStoreService = service_evidenceStore.NewService()
 
+	assessmentService.RegisterAssessmentResultHook(func(result *assessment.AssessmentResult, err error) {
```

Comment:
Just for my understanding: Why do we need an `error` as parameter of the hook function? Maybe just state a short example. Thanks!

Reply:
Depending on the previous code, you may already have an error and want to handle it differently. See service/assessment.go line 117 (updated file).

Reply:
I see. Probably in our case, the hook should not be called as well when we do not accept the evidence. But I can imagine that there are proper use cases.